src/CRISP.BackendChallenge/Controllers/EmployeeController.cs

```diff
@@ -7,33 +7,266 @@
+            var employees = _employeeRepository.GetAll();
+
+            var result = employees.Select(ToEmployeeResponse);
+
+            return Ok(result);
+        }
+        catch (ArgumentNullException e)
+        {
+            _logger.LogDebug(":: There are no Employees in the Database: {EMessage}", e.Message);
+            return NotFound(new {error = "No Employees Found in Database"});
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(":: Error occurred while getting all employees: {EMessage}", e.Message);
+            return CreateInternalServerError();
+        }
+    }
+
+    [HttpGet]
+    [Route("{id:int}")]
+    public IActionResult GetById(int id)
+    {
+        _logger.LogDebug(":: Performing {MethodName} with ID {Id}", nameof(GetById), id);
+        try
+        {
+            var employee = _employeeRepository.GetById(id);
+            return Ok(ToEmployeeResponse(employee));
+        }
+        catch (ArgumentNullException e)
+        {
+            _logger.LogDebug(":: Id Not Found: {Id}, {EMessage}", id, e.Message);
+            return CreateNotFound(id);
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(":: Error occurred while getting employee ID: {Id} {EMessage}", id, e.Message);
+            return CreateInternalServerError();
+        }
+    }
+    
+    
+    [HttpPost]
+    [Route("createEmployee")]
+    public IActionResult AddEmployee([FromBody] EmployeeRequest employeeRequest)
+    {
+        _logger.LogDebug(":: Performing {MethodName}", nameof(AddEmployee));
+        
+        if (string.IsNullOrWhiteSpace(employeeRequest.Name))
+        {
+            _logger.LogError(":: Invalid EmployeeRequest provided");
+            return CreateBadRequest("Creating an employee", "Name field required");
+        }
+        if (employeeRequest.Department == null)
+        {
+            _logger.LogError(":: Invalid EmployeeRequest provided");
+            return CreateBadRequest("Creating an employee", "Department field required");
+        }
+        
+        try
+        {
+            
+            var employee = new Employee
+            {
+                Name = employeeRequest.Name,
+                Department = (Department)employeeRequest.Department,
+                StartDate = employeeRequest.StartDate,
+                EndDate = employeeRequest.EndDate
+            };
+
+            _employeeRepository.Add(employee);
+            _employeeRepository.Save();
+
+            if (employeeRequest.LoginDates != null || employeeRequest.LoginDates!.Any())
+            {
+                foreach (var login in employeeRequest.LoginDates!.Select(l => new Login()
+                         {
+                             EmployeeId = employee.Id,
+                             LoginDate = l
+                         }))
+                {
+                    _loginRepository.Add(login);
+                }
+                _loginRepository.Save();
+            }
```

Comment:
A general thing which also applies elsewhere - this sort of logic does not belong in the controller. If we are following the Repository pattern, your repository would be responsible for encapsulating all query/command logic such as this. The controller's job is to:

1. Provide a declarative definition of the endpoints
2. Validate the request (depending on your validation strategy it might not even do this)
3. Determine the proper response status code and response to return to the user based on the end result of the operation

Reply:
👍🏼